statsmodels/regression/linear_model.py

```diff
@@ -816,21 +819,44 @@ def iterative_fit(self, maxiter=3):
         ----------
         maxiter : integer, optional
             the number of iterations
+        rtol : float, optional
+            Relative tolerance between estimated coefficients to stop the
+            estimation.  Stops if
+
+            max(abs(last - current) / abs(last)) < rtol
+
         """
-        #TODO: update this after going through example.
-        for i in range(maxiter-1):
+        # TODO: update this after going through example.
+        history = {'params': []}
+        for i in range(maxiter - 1):
             if hasattr(self, 'pinv_wexog'):
                 del self.pinv_wexog
             self.initialize()
             results = self.fit()
+            history['params'].append(results.params)
+            if i == 0:
+                last = results.params
+            else:
+                diff = np.max(np.abs(last - results.params) / np.abs(last))
+                if diff < rtol:
+                    break
+                last = results.params
             self.rho, _ = yule_walker(results.resid,
                                       order=self.order, df=None)
-        #why not another call to self.initialize
+        # why not another call to self.initialize
         if hasattr(self, 'pinv_wexog'):
             del self.pinv_wexog
         self.initialize()
-        results = self.fit() #final estimate
-        return results # add missing return
+        if _is_using_pandas(history['params'][0], None):
```

Comment:
@bashtage  Was there a specific reason to converting the list, besides having a nicer print?
I'm planning on removing this for 0.7, because it make indexing difficult because of inconsistent behavior between pandas, numpy, and python lists.

A possible alternative would be to use a simple history subclass of python list that has a `as_dataframe` attribute, which could also add rho and maybe other things to the history DataFrame. But this would have to be introduced consistently for all histories in models with specialized iterative solvers.